File: cmd/chopsticks/cli/template/bucket-lua/apps/tools.lua
local M = {}

function M.trim(s)
    return s:match("^%s*(.-)%s*$")
end

function M.split(s, delimiter)
    local result = {}
    for match in string.gmatch(s, "([^" .. delimiter .. "]+)") do
        table.insert(result, match)
    end
    return result
end

function M.starts_with(str, prefix)
    return string.sub(str, 1, string.len(prefix)) == prefix
end

function M.ends_with(str, suffix)
    return suffix == "" or string.sub(str, -string.len(suffix)) == suffix
end

return M

File: cmd/cli/template/bucket-js/apps/_tools_.js
/**
 * @global
 * @param {string} s
 * @returns {string}
 */
let trim = (s) => s.trim();

/**
 * @global
 * @param {string} s
 * @param {string} delimiter
 * @returns {string[]}
 */
let split = (s, delimiter) => {
    return s.split(delimiter);
};

/**
 * @global
 * @param {string} str
 * @param {string} prefix
 * @returns {boolean}
 */
let startsWith = (str, prefix) => {
    return str.startsWith(prefix);
};

/**
 * @global
 * @param {string} str
 * @param {string} suffix
 * @returns {boolean}
 */
let endsWith = (str, suffix) => {
    return str.endsWith(suffix);
};
